src/Integration.Vsix/SonarLintTagger/IssueTagger.cs

```diff
@@ -37,21 +37,21 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     /// </remarks>
     internal sealed class IssueTagger : ITagger<IErrorTag>, IDisposable
     {
-        private IssuesSnapshot issues;
+        public delegate void OnTaggerDisposed(IssueTagger issueTagger);
+
+        private IEnumerable<IssueMarker> issueMarkers;
+        private readonly OnTaggerDisposed onTaggerDisposed;
         private bool isDisposed;
-        internal /* for testing */ IIssueTracker IssueTracker { get; }
 
-        public IssueTagger(IIssueTracker issueTracker)
+        public IssueTagger(IEnumerable<IssueMarker> issueMarkers, OnTaggerDisposed onTaggerDisposed)
         {
-            this.IssueTracker = issueTracker;
-            this.issues = issueTracker.LastIssues;
-
-            issueTracker.AddTagger(this);
+            this.issueMarkers = issueMarkers;
+            this.onTaggerDisposed = onTaggerDisposed;
         }
 
-        public void UpdateMarkers(IssuesSnapshot newIssues, SnapshotSpan? affectedSpan)
+        public void UpdateMarkers(IEnumerable<IssueMarker> newMarkers, SnapshotSpan? affectedSpan)
```

Comment:
Simplification: the tagger doesn't need to know about the `IssuesSnapshot` either. It only ever accessed the list of `IssueMarkers`.